src/components/codeContext.tsx

```diff
@@ -277,10 +284,21 @@ export async function createOrgAuthToken({
   }
 }
 
+const getLocallyStoredSelections = (): SelectedCodeTabs => {
+  if (isLocalStorageAvailable()) {
+    return Object.fromEntries(
+      Object.entries(localStorage).filter(([key]) => key.startsWith('Tabgroup:'))
+    );
+  }
+  return {};
+};
+
 export function CodeContextProvider({children}: {children: React.ReactNode}) {
   const [codeKeywords, setCodeKeywords] = useState(cachedCodeKeywords ?? DEFAULTS);
-
   const [isLoading, setIsLoading] = useState<boolean>(cachedCodeKeywords ? false : true);
+  const [storedCodeSelection, setStoredCodeSelection] = useState<SelectedCodeTabs>(() =>
+    getLocallyStoredSelections()
```

Comment:
l/m: As discussed: I would not make the initial state dependent on localStorage as it invites hydration errors in the future.

Reply:
updated it 👍